include/glob.hpp

```diff
@@ -28,47 +29,82 @@ namespace glob {
 
 		/* Add */
 		vect operator+(const vect &v) const { return vect(x + v.x,
-		                                                  y + v.y); };
+														  y + v.y); };
 
 		/* Subtract */
 		vect operator-(const vect &v) const { return vect(x - v.x,
-		                                                  y - v.y); };
+														  y - v.y); };
 
 		/* Multiply by a scalar */
 		vect operator*(const float &s) const { return vect(x * s,
-		                                                   y * s); };
+														   y * s); };
 
 		/* Divide by a scalar */
 		vect operator/(const float &s) const { return vect(x / s,
-		                                                   y / s); };
+														   y / s); };
 
 		/* Dot product */
-		float operator*(const vect &v) const { return x * v.x
-		                                            + y * v.y; };
+		float operator*(const vect &v) const { return x * v.x + y * v.y; };
 
 		/* Equals? */
 		bool operator==(const vect &v) const { return x == v.x && y == v.y; };
 
 		/* Subtract from this vector */
-		void operator+=(const vect &v) { x += v.x; y += v.y; };
+		void operator+=(const vect &v)
+		{
+			x += v.x;
+			y += v.y;
+		};
 
 		/* Subtract from this vector */
-		void operator-=(const vect &v) { x -= v.x; y -= v.y; };
+		void operator-=(const vect &v)
+		{
+			x -= v.x;
+			y -= v.y;
+		};
 
 		/* Multiply into this vector by a scalar */
-		void operator*=(const float &s) { x *= s; y *= s; };
+		void operator*=(const float &s)
+		{
+			x *= s;
+			y *= s;
+		};
+
+		/* Entry-wise multiply into this vector by another vector */
+		void operator*=(const vect &v)
+		{
+			x *= v.x;
+			y *= v.y;
+		}
 
 		/* Divide into this vector by a scalar */
-		void operator/=(const float &s) { x /= s; y /= s; };
+		void operator/=(const float &s)
+		{
+			x /= s;
+			y /= s;
+		};
 
 		float x, y;
 	};
 
 	/* Multiply by a scalar */
-	inline vect operator*(const float &s, const vect &v) {
+	inline vect operator*(const float &s, const vect &v)
+	{
 		return vect(v.x * s, v.y * s);
 	};
 
+	inline vect zeroVector = vect(0, 0);
+
+	/**
+ 	 * Convert vector into a unit vector
+ 	*/
+	inline vect normalize(const vect &v)
+	{
+		if(v == zeroVector) return v; 
```

Comment:
I'm not sure why we need a zero vector in the glob namespace, you could just check `v == { 0, 0 }`? If you insist on adding it, at least make it snake case!